### ISSUE TITLE
Cordova Android start failed with Uncaught Error because of adding accounts-base

### ISSUE THREAD

**Issue by joyjiyuan** [meteor/meteor#13450]
Meteor: 3.0.4
Cordova Android: 13.0.0
OS: both windows 11 and macOS 15
Emulator: Pixel 8a API 34

### The expected behavior
Run on android emulator or device
### The actual behavior.
Start failed after add accounts-base. white screen
### reproduce
1. create meteor app from skelton
2. meteor add accounts-base
3. meteor add-platform android
4. meteor run android

"Uncaught Error: The core-js npm package could not be found in your node_modules
...
Uncaught TypeError: Cannot read properties of undefined (reading 'Accounts')"

and run meteor npm install --save core-js doesn't fixed it

I20241108-04:27:19.445(9)? 11-07 19:27:05.028  9188  9188 I CordovaLog: Changing log level to DEBUG(3)
I20241108-04:27:19.460(9)? 11-07 19:27:06.501  9188  9320 W chromium: [WARNING:dns_config_service_android.cc(115)] Failed to read DnsConfig.
I20241108-04:27:19.461(9)? 11-07 19:27:07.824  9188  9188 W MeteorWebApp: Loading asset bundle from directory file:///android_asset/www/application
I20241108-04:27:19.461(9)? 11-07 19:27:08.453  9188  9320 W MeteorWebApp: Asset / found in bundle 5475bb95b63169f6f68ab99e9362c96538c39475:file:///android_asset/www/application
I20241108-04:27:19.461(9)? 11-07 19:27:08.734  9188  9319 W MeteorWebApp: Asset /__cordova/merged-stylesheets.css found in bundle 5475bb95b63169f6f68ab99e9362c96538c39475:file:///android_asset/www/application
I20241108-04:27:19.461(9)? 11-07 19:27:08.987  9188  9319 W MeteorWebApp: Asset /cordova.js not found in bundle 5475bb95b63169f6f68ab99e9362c96538c39475:file:///android_asset/www/application, no parent bundle
I20241108-04:27:19.461(9)? 11-07 19:27:09.114  9188  9319 W MeteorWebApp: Asset /cordova_plugins.js not found in bundle 5475bb95b63169f6f68ab99e9362c96538c39475:file:///android_asset/www/application, no parent bundle
I20241108-04:27:19.461(9)? 11-07 19:27:09.115  9188  9320 W MeteorWebApp: Asset /__cordova/packages/core-runtime.js found in bundle 5475bb95b63169f6f68ab99e9362c96538c39475:file:///android_asset/www/application
I20241108-04:27:19.461(9)? 11-07 19:27:09.231  9188  9319 W MeteorWebApp: Asset /plugins/cordova-plugin-meteor-webapp/www/webapp_local_server.js not found in bundle 5475bb95b63169f6f68ab99e9362c96538c39475:file:///android_asset/www/application, no parent bundle
I20241108-04:27:19.461(9)? 11-07 19:27:09.231  9188  9320 W MeteorWebApp: Asset /plugins/cordova-plugin-statusbar/www/statusbar.js not found in bundle 5475bb95b63169f6f68ab99e9362c96538c39475:file:///android_asset/www/application, no parent bundle
I20241108-04:27:19.461(9)? 11-07 19:27:09.236  9188  9321 W MeteorWebApp: Asset /__cordova/packages/meteor.js found in bundle 5475bb95b63169f6f68ab99e9362c96538c39475:file:///android_asset/www/application
I20241108-04:27:19.461(9)? 11-07 19:27:09.276  9188  9319 W MeteorWebApp: Asset /plugins/cordova-plugin-splashscreen/www/splashscreen.js not found in bundle 5475bb95b63169f6f68ab99e9362c96538c39475:file:///android_asset/www/application, no parent bundle   
I20241108-04:27:19.461(9)? 11-07 19:27:09.342  9188  9319 W MeteorWebApp: Asset /__cordova/packages/meteor-base.js found in bundle 5475bb95b63169f6f68ab99e9362c96538c39475:file:///android_asset/www/application
I20241108-04:27:19.461(9)? 11-07 19:27:09.387  9188  9319 W MeteorWebApp: Asset /__cordova/packages/mobile-experience.js found in bundle 5475bb95b63169f6f68ab99e9362c96538c39475:file:///android_asset/www/application
I20241108-04:27:19.461(9)? 11-07 19:27:09.423  9188  9319 W MeteorWebApp: Asset /__cordova/packages/modules-runtime.js found in bundle 5475bb95b63169f6f68ab99e9362c96538c39475:file:///android_asset/www/application
I20241108-04:27:19.461(9)? 11-07 19:27:09.446  9188  9319 W MeteorWebApp: Asset /__cordova/packages/modules-runtime-hot.js found in bundle 5475bb95b63169f6f68ab99e9362c96538c39475:file:///android_asset/www/application
I20241108-04:27:19.461(9)? 11-07 19:27:09.478  9188  9319 W MeteorWebApp: Asset /__cordova/packages/modules.js found in bundle 5475bb95b63169f6f68ab99e9362c96538c39475:file:///android_asset/www/application
I20241108-04:27:19.461(9)? 11-07 19:27:09.583  9188  9319 W MeteorWebApp: Asset /__cordova/packages/es5-shim.js found in bundle 5475bb95b63169f6f68ab99e9362c96538c39475:file:///android_asset/www/application
I20241108-04:27:19.461(9)? 11-07 19:27:09.658  9188  9319 W MeteorWebApp: Asset /__cordova/packages/modern-browsers.js found in bundle 5475bb95b63169f6f68ab99e9362c96538c39475:file:///android_asset/www/application
I20241108-04:27:19.462(9)? 11-07 19:27:09.683  9188  9319 W MeteorWebApp: Asset /__cordova/packages/promise.js found in bundle 5475bb95b63169f6f68ab99e9362c96538c39475:file:///android_asset/www/application
I20241108-04:27:19.462(9)? 11-07 19:27:09.710  9188  9319 W MeteorWebApp: Asset /__cordova/packages/ecmascript-runtime-client.js found in bundle 5475bb95b63169f6f68ab99e9362c96538c39475:file:///android_asset/www/application
I20241108-04:27:19.462(9)? 11-07 19:27:09.857  9188  9319 W MeteorWebApp: Asset /__cordova/packages/hot-module-replacement.js found in bundle 5475bb95b63169f6f68ab99e9362c96538c39475:file:///android_asset/www/application
I20241108-04:27:19.462(9)? 11-07 19:27:09.885  9188  9320 W MeteorWebApp: Asset /__cordova/packages/react-fast-refresh.js found in bundle 5475bb95b63169f6f68ab99e9362c96538c39475:file:///android_asset/www/application
I20241108-04:27:19.462(9)? 11-07 19:27:09.939  9188  9188 I chromium: [INFO:CONSOLE(341)] "HMR: connected", source: http://localhost:12792/__cordova/packages/hot-module-replacement.js?hash=be8c9e0d70af1086ff2c8f2fb6109ea31b6ff6b0 (341)
I20241108-04:27:19.462(9)? 11-07 19:27:14.045  9188  9188 I chromium: [INFO:CONSOLE(1237)] "deviceready has not fired after 5 seconds.", source: http://localhost:12792/cordova.js (1237)
I20241108-04:27:19.462(9)? 11-07 19:27:14.045  9188  9188 I chromium: [INFO:CONSOLE(1230)] "Channel not fired: onDOMContentLoaded", source: http://localhost:12792/cordova.js (1230)
I20241108-04:27:19.462(9)? 11-07 19:27:14.901  9188  9319 W MeteorWebApp: Asset /__cordova/packages/ecmascript.js found in bundle 5475bb95b63169f6f68ab99e9362c96538c39475:file:///android_asset/www/application
I20241108-04:27:19.462(9)? 11-07 19:27:14.921  9188  9319 W MeteorWebApp: Asset /__cordova/packages/ecmascript-runtime.js found in bundle 5475bb95b63169f6f68ab99e9362c96538c39475:file:///android_asset/www/application
I20241108-04:27:19.462(9)? 11-07 19:27:14.958  9188  9319 W MeteorWebApp: Asset /__cordova/packages/babel-runtime.js found in bundle 5475bb95b63169f6f68ab99e9362c96538c39475:file:///android_asset/www/application
I20241108-04:27:19.462(9)? 11-07 19:27:14.994  9188  9319 W MeteorWebApp: Asset /__cordova/packages/fetch.js found in bundle 5475bb95b63169f6f68ab99e9362c96538c39475:file:///android_asset/www/application
I20241108-04:27:19.462(9)? 11-07 19:27:15.038  9188  9319 W MeteorWebApp: Asset /__cordova/packages/dynamic-import.js found in bundle 5475bb95b63169f6f68ab99e9362c96538c39475:file:///android_asset/www/application
I20241108-04:27:19.462(9)? 11-07 19:27:15.070  9188  9319 W MeteorWebApp: Asset /__cordova/packages/base64.js found in bundle 5475bb95b63169f6f68ab99e9362c96538c39475:file:///android_asset/www/application
I20241108-04:27:19.462(9)? 11-07 19:27:15.102  9188  9319 W MeteorWebApp: Asset /__cordova/packages/ejson.js found in bundle 5475bb95b63169f6f68ab99e9362c96538c39475:file:///android_asset/www/application
I20241108-04:27:19.462(9)? 11-07 19:27:15.138  9188  9319 W MeteorWebApp: Asset /__cordova/packages/diff-sequence.js found in bundle 5475bb95b63169f6f68ab99e9362c96538c39475:file:///android_asset/www/application
I20241108-04:27:19.462(9)? 11-07 19:27:15.170  9188  9319 W MeteorWebApp: Asset /__cordova/packages/geojson-utils.js found in bundle 5475bb95b63169f6f68ab99e9362c96538c39475:file:///android_asset/www/application
I20241108-04:27:19.462(9)? 11-07 19:27:15.203  9188  9319 W MeteorWebApp: Asset /__cordova/packages/id-map.js found in bundle 5475bb95b63169f6f68ab99e9362c96538c39475:file:///android_asset/www/application
I20241108-04:27:19.462(9)? 11-07 19:27:15.245  9188  9319 W MeteorWebApp: Asset /__cordova/packages/random.js found in bundle 5475bb95b63169f6f68ab99e9362c96538c39475:file:///android_asset/www/application
I20241108-04:27:19.462(9)? 11-07 19:27:15.288  9188  9319 W MeteorWebApp: Asset /__cordova/packages/mongo-id.js found in bundle 5475bb95b63169f6f68ab99e9362c96538c39475:file:///android_asset/www/application
I20241108-04:27:19.462(9)? 11-07 19:27:15.327  9188  9319 W MeteorWebApp: Asset /__cordova/packages/ordered-dict.js found in bundle 5475bb95b63169f6f68ab99e9362c96538c39475:file:///android_asset/www/application
I20241108-04:27:19.462(9)? 11-07 19:27:15.366  9188  9319 W MeteorWebApp: Asset /__cordova/packages/tracker.js found in bundle 5475bb95b63169f6f68ab99e9362c96538c39475:file:///android_asset/www/application
I20241108-04:27:19.462(9)? 11-07 19:27:15.405  9188  9319 W MeteorWebApp: Asset /__cordova/packages/minimongo.js found in bundle 5475bb95b63169f6f68ab99e9362c96538c39475:file:///android_asset/www/application
I20241108-04:27:19.462(9)? 11-07 19:27:15.486  9188  9319 W MeteorWebApp: Asset /__cordova/packages/check.js found in bundle 5475bb95b63169f6f68ab99e9362c96538c39475:file:///android_asset/www/application
I20241108-04:27:19.462(9)? 11-07 19:27:15.525  9188  9319 W MeteorWebApp: Asset /__cordova/packages/retry.js found in bundle 5475bb95b63169f6f68ab99e9362c96538c39475:file:///android_asset/www/application
I20241108-04:27:19.462(9)? 11-07 19:27:15.557  9188  9319 W MeteorWebApp: Asset /__cordova/packages/callback-hook.js found in bundle 5475bb95b63169f6f68ab99e9362c96538c39475:file:///android_asset/www/application
I20241108-04:27:19.462(9)? 11-07 19:27:15.594  9188  9319 W MeteorWebApp: Asset /__cordova/packages/ddp-common.js found in bundle 5475bb95b63169f6f68ab99e9362c96538c39475:file:///android_asset/www/application
I20241108-04:27:19.462(9)? 11-07 19:27:15.630  9188  9319 W MeteorWebApp: Asset /__cordova/packages/reload.js found in bundle 5475bb95b63169f6f68ab99e9362c96538c39475:file:///android_asset/www/application
I20241108-04:27:19.462(9)? 11-07 19:27:15.678  9188  9319 W MeteorWebApp: Asset /__cordova/packages/socket-stream-client.js found in bundle 5475bb95b63169f6f68ab99e9362c96538c39475:file:///android_asset/www/application
I20241108-04:27:19.462(9)? 11-07 19:27:15.721  9188  9319 W MeteorWebApp: Asset /__cordova/packages/ddp-client.js found in bundle 5475bb95b63169f6f68ab99e9362c96538c39475:file:///android_asset/www/application
I20241108-04:27:19.462(9)? 11-07 19:27:15.849  9188  9319 W MeteorWebApp: Asset /__cordova/packages/ddp.js found in bundle 5475bb95b63169f6f68ab99e9362c96538c39475:file:///android_asset/www/application
I20241108-04:27:19.462(9)? 11-07 19:27:15.890  9188  9319 W MeteorWebApp: Asset /__cordova/packages/ddp-server.js found in bundle 5475bb95b63169f6f68ab99e9362c96538c39475:file:///android_asset/www/application
I20241108-04:27:19.462(9)? 11-07 19:27:15.918  9188  9319 W MeteorWebApp: Asset /__cordova/packages/allow-deny.js found in bundle 5475bb95b63169f6f68ab99e9362c96538c39475:file:///android_asset/www/application
I20241108-04:27:19.462(9)? 11-07 19:27:15.960  9188  9319 W MeteorWebApp: Asset /__cordova/packages/mongo-dev-server.js found in bundle 5475bb95b63169f6f68ab99e9362c96538c39475:file:///android_asset/www/application
I20241108-04:27:19.462(9)? 11-07 19:27:16.012  9188  9319 W MeteorWebApp: Asset /__cordova/packages/babel-compiler.js found in bundle 5475bb95b63169f6f68ab99e9362c96538c39475:file:///android_asset/www/application
I20241108-04:27:19.462(9)? 11-07 19:27:16.035  9188  9319 W MeteorWebApp: Asset /__cordova/packages/typescript.js found in bundle 5475bb95b63169f6f68ab99e9362c96538c39475:file:///android_asset/www/application
I20241108-04:27:19.475(9)? 11-07 19:27:16.062  9188  9319 W MeteorWebApp: Asset /__cordova/packages/logging.js found in bundle 5475bb95b63169f6f68ab99e9362c96538c39475:file:///android_asset/www/application
I20241108-04:27:19.497(9)? 11-07 19:27:16.087  9188  9319 W MeteorWebApp: Asset /__cordova/packages/mongo.js found in bundle 5475bb95b63169f6f68ab99e9362c96538c39475:file:///android_asset/www/application
I20241108-04:27:19.539(9)? 11-07 19:27:16.133  9188  9319 W MeteorWebApp: Asset /__cordova/packages/reactive-var.js found in bundle 5475bb95b63169f6f68ab99e9362c96538c39475:file:///android_asset/www/application
I20241108-04:27:19.558(9)? 11-07 19:27:16.151  9188  9319 W MeteorWebApp: Asset /__cordova/packages/minifier-css.js found in bundle 5475bb95b63169f6f68ab99e9362c96538c39475:file:///android_asset/www/application
I20241108-04:27:19.581(9)? 11-07 19:27:16.174  9188  9319 W MeteorWebApp: Asset /__cordova/packages/standard-minifier-css.js found in bundle 5475bb95b63169f6f68ab99e9362c96538c39475:file:///android_asset/www/application
I20241108-04:27:19.601(9)? 11-07 19:27:16.194  9188  9319 W MeteorWebApp: Asset /__cordova/packages/standard-minifier-js.js found in bundle 5475bb95b63169f6f68ab99e9362c96538c39475:file:///android_asset/www/application
I20241108-04:27:19.623(9)? 11-07 19:27:16.215  9188  9319 W MeteorWebApp: Asset /__cordova/packages/shell-server.js found in bundle 5475bb95b63169f6f68ab99e9362c96538c39475:file:///android_asset/www/application
I20241108-04:27:19.649(9)? 11-07 19:27:16.242  9188  9319 W MeteorWebApp: Asset /__cordova/packages/static-html.js found in bundle 5475bb95b63169f6f68ab99e9362c96538c39475:file:///android_asset/www/application
I20241108-04:27:19.668(9)? 11-07 19:27:16.260  9188  9319 W MeteorWebApp: Asset /__cordova/packages/react-meteor-data.js found in bundle 5475bb95b63169f6f68ab99e9362c96538c39475:file:///android_asset/www/application
I20241108-04:27:19.685(9)? 11-07 19:27:16.278  9188  9319 W MeteorWebApp: Asset /__cordova/packages/ddp-rate-limiter.js found in bundle 5475bb95b63169f6f68ab99e9362c96538c39475:file:///android_asset/www/application
I20241108-04:27:19.704(9)? 11-07 19:27:16.296  9188  9319 W MeteorWebApp: Asset /__cordova/packages/localstorage.js found in bundle 5475bb95b63169f6f68ab99e9362c96538c39475:file:///android_asset/www/application
I20241108-04:27:19.725(9)? 11-07 19:27:16.318  9188  9319 W MeteorWebApp: Asset /__cordova/packages/url.js found in bundle 5475bb95b63169f6f68ab99e9362c96538c39475:file:///android_asset/www/application
**_I20241108-04:27:19.791(9)? 11-07 19:27:16.384  9188  9188 I chromium: [INFO:CONSOLE(32)] "Uncaught Error: The core-js npm package could not be found in your node_modules
I20241108-04:27:19.791(9)? 11-07 19:27:16.384  9188  9188 I chromium: directory. Please run the following command to install it:
I20241108-04:27:19.792(9)? 11-07 19:27:16.384  9188  9188 I chromium:
I20241108-04:27:19.792(9)? 11-07 19:27:16.384  9188  9188 I chromium:   meteor npm install --save core-js
I20241108-04:27:19.792(9)? 11-07 19:27:16.384  9188  9188 I chromium: ", source: http://localhost:12792/__cordova/packages/url.js?hash=7f37812133a7907a1ca39558aad174b72ed4ecc9 (32)_**
I20241108-04:27:19.794(9)? 11-07 19:27:16.385  9188  9319 W MeteorWebApp: Asset /__cordova/packages/accounts-base.js found in bundle 5475bb95b63169f6f68ab99e9362c96538c39475:file:///android_asset/www/application
I20241108-04:27:19.812(9)? 11-07 19:27:16.404  9188  9319 W MeteorWebApp: Asset /__cordova/packages/webapp.js found in bundle 5475bb95b63169f6f68ab99e9362c96538c39475:file:///android_asset/www/application
I20241108-04:27:19.835(9)? 11-07 19:27:16.428  9188  9319 W MeteorWebApp: Asset /__cordova/packages/hot-code-push.js found in bundle 5475bb95b63169f6f68ab99e9362c96538c39475:file:///android_asset/www/application
I20241108-04:27:19.853(9)? 11-07 19:27:16.447  9188  9319 W MeteorWebApp: Asset /__cordova/packages/mobile-status-bar.js found in bundle 5475bb95b63169f6f68ab99e9362c96538c39475:file:///android_asset/www/application
I20241108-04:27:19.870(9)? 11-07 19:27:16.463  9188  9319 W MeteorWebApp: Asset /__cordova/packages/launch-screen.js found in bundle 5475bb95b63169f6f68ab99e9362c96538c39475:file:///android_asset/www/application
I20241108-04:27:19.890(9)? 11-07 19:27:16.483  9188  9319 W MeteorWebApp: Asset /__cordova/packages/autoupdate.js found in bundle 5475bb95b63169f6f68ab99e9362c96538c39475:file:///android_asset/www/application
I20241108-04:27:19.918(9)? 11-07 19:27:16.511  9188  9319 W MeteorWebApp: Asset /__cordova/app/global-imports.js found in bundle 5475bb95b63169f6f68ab99e9362c9I20241108-04:27:19.937(9)? 11-07 19:27:16.531  9188  9319 W MeteorWebApp: Asset /__cordova/app/app.js found in bundle 5475bb95b63169f6f68ab99e9362c96538c39475:file:///android_asset/www/application
**_I20241108-04:27:19.940(9)? 11-07 19:27:16.532  9188  9188 I chromium: [INFO:CONSOLE(5)] "Uncaught TypeError: Cannot read properties of undefined (reading 'Accounts')", source: http://localhost:12792/__cordova/app/global-imports.js?hash=cc5a508356a07ac1a5b71e636309c05b05e50183 (5)
I20241108-04:27:19.948(9)? 11-07 19:27:16.540  9188  9188 I chromium: [INFO:CONSOLE(1)] "Uncaught ReferenceError: meteorInstall is not defined", source: http://localhost:12792/__cordova/app/app.js?hash=327d5fdbe10ef293155948a8541105c45f0fd6bf (1)_**
I20241108-04:27:19.976(9)? 11-07 19:27:16.569  9188  9319 W MeteorWebApp: Asset /favicon.ico not found in bundle 5475bb95b63169f6f68ab99e9362c96538c39475:file:///android_asset/www/application, no parent bundle
I20241108-04:27:19.982(9)? 11-07 19:27:16.575  9188  9319 E SystemWebViewClient: www/favicon.ico

**Comment by MarcoTribuz** [comment]
I have the same error on ios platform! :)

**Comment by MarcoTribuz** [comment]
I noticed that in the package url on legacy js file, i have this kind of error

`TypeError: undefined is not an object (evaluating '$URLSearchParams.prototype')`

**Comment by MarcoTribuz** [comment]
It seems that cordova use the legacy.js file to render the content, i tried to comment the line in package.js api.mainModule("legacy.js", "legacy"), and the problem is solved, biut i dont think it is the best approach. But we can start our investigation on why cordova use legacy.js of url package...

**Comment by joyjiyuan** [comment]
> It seems that cordova use the legacy.js file to render the content, i tried to comment the line in package.js api.mainModule("legacy.js", "legacy"), and the problem is solved, biut i dont think it is the best approach. But we can start our investigation on why cordova use legacy.js of url package...

It is indeed related to URL and legacy.
I found this section in the generated file.

```
try {
  require("core-js/proposals/url");
} catch (e) {
  throw new Error([
    "The core-js npm package could not be found in your node_modules ",
    "directory. Please run the following command to install it:",
    "",
    "  meteor npm install --save core-js",
    ""
  ].join("\n"));
}

// backwards compatibility
require('./modern.js');
```

But how to eliminate this problem. I can't find the part you metioned. api.mainModule("legacy.js", "legacy")

**Comment by MarcoTribuz** [comment]
Hi [joyjiyuan](https://github.com/joyjiyuan)!

![Screenshot 2024-11-10 at 09 14 12](https://github.com/user-attachments/assets/f89bada5-8ae3-4528-9023-3a4ddcebe848)

as the image, you have to have a folder named packages in your project, than clone the url package as you can see in the image, and then in package.js

`Package.describe({
  name: "url",
  version: '1.3.4',
  summary: "Isomorphic modern/legacy/Node polyfill for WHATWG URL/URLSearchParams",
  documentation: "README.md"
});

Npm.depends({
  "core-js": "3.39.0"
});

Package.onUse(function(api) {
  api.use("modules");
  api.use("modern-browsers");

  api.mainModule("modern.js", "web.browser");
  //api.mainModule("legacy.js", "legacy"); <-- here i commented out
  api.mainModule("server.js", "server");

  api.export("URL");
  api.export("URLSearchParams");
});

Package.onTest(function(api) {
  api.use("ecmascript");
  api.use("tinytest");
  api.use("url");
  api.mainModule("tests/main.js");
});`

as is said, i don't know if it is the best approach, but at least i can continue the development of my app :)

**Comment by joyjiyuan** [comment]
Cool, it worked. Now I can move on. Thanks. @MarcoTribuz

**Comment by nachocodoner** [comment]
Does anyone know if this has happened before 3.0.4 by a chance?

I will look into it for next release, but I would like to know if this is a regression, or we have been living with this from early Meteor 3 versions.

Thank you for the report.

__Update: I reproduced the problem. This happens as well on Meteor 3.0.1. So this is definitely an issue from early versions of Meteor 3, not a regression recently introduced. We will work on the fix.__

**Comment by nachocodoner** [comment]
https://github.com/meteor/meteor/pull/13459

I’ve found the solution. It may be safe to include it under 3.1, but I’ll need to test if it validates proper account behavior across real apps after this fix. Could anyone assist?

If I get your feedback we can more safely include it on 3.1.

**Comment by MarcoTribuz** [comment]
I'm in! Tell me what I have to do. It will be a pleasure! 😊

**Comment by nachocodoner** [comment]
Could you fork the Meteor repository, check out the branch with the fix, and test the app with the issue?

## Steps

1) Clone the Meteor project: `git clone https://github.com/meteor/meteor.git`

2) Switch to the fix branch: `git checkout fix-cordova-accounts-url-legacy`

3) Run your cordova app from the meteor checkout: `<path-to-meteor-checkout>/meteor run ios`

Thank you so much!

**Comment by MarcoTribuz** [comment]
Ok! I'll try tomorrow and I'll keep you posted! 😊🥳

**Comment by MarcoTribuz** [comment]
Hi [nachocodoner](https://github.com/nachocodoner)!

Sorry for the delay, i tried your code this morning but the project doesn't start at all, you can see the log attached below

To give you a bit of context, i'm using a mac studio m1, with webstorm, macos sequoia :)

UPDATE and i tried this command too but no success:

    "start": "ARCH=arm64 METEOR_PORT=3000 ROOT_URL=http://192.168.10.202 MOBILE_ROOT_URL=http://192.168.10.202 /Users/marcotribuzio/WebstormProjects/meteor/meteor run ios-device --mobile-server http://192.168.10.202:3000 --port 3000",

(node:18628) Warning: The `util._extend` API is deprecated. Please use Object.assign() instead.
(Use `node --trace-warnings ...` to show where the warning was created)
failed to load config from /Users/marcotribuzio/WebstormProjects/simple-todos-vue/vite.config.js
⚡  [vite-dev-server] Failed to handle IPC request {
  method: 'vite.server.start',
  params: '[{"packageJson":{"name":"simple-todos-vue","private":true,"scripts":{"start":"meteor run ios-device --mobile-server http://localhost:3000 --por
} Error: 
You installed esbuild for another platform than the one you're currently using.
This won't work because esbuild is written with native code and needs to
install a platform-specific binary executable.
Specifically the "esbuild-darwin-64" package is present but this platform
needs the "esbuild-darwin-arm64" package instead. People often get into this
situation by installing esbuild with npm running inside of Rosetta 2 and then
trying to use it with node running outside of Rosetta 2, or vice versa (Rosetta
2 is Apple's on-the-fly x86_64-to-arm64 translation service).
If you are installing with npm, you can try ensuring that both npm and node are
not running under Rosetta 2 and then reinstalling esbuild. This likely involves
changing how you installed npm and/or node. For example, installing node with
the universal installer here should work: https://nodejs.org/en/download/. Or
you could consider using yarn instead of npm which has built-in support for
installing a package on multiple platforms simultaneously.
If you are installing with yarn, you can try listing both "arm64" and "x64"
in your ".yarnrc.yml" file using the "supportedArchitectures" feature:
https://yarnpkg.com/configuration/yarnrc/#supportedArchitectures
Keep in mind that this means multiple copies of esbuild will be present.
Another alternative is to use the "esbuild-wasm" package instead, which works
the same way on all platforms. But it comes with a heavy performance cost and
can sometimes be 10x slower than the "esbuild" package, so you may also not
want to do that.
    at generateBinPath (/Users/marcotribuzio/WebstormProjects/simple-todos-vue/node_modules/esbuild/lib/main.js:1808:17)
    at esbuildCommandAndArgs (/Users/marcotribuzio/WebstormProjects/simple-todos-vue/node_modules/esbuild/lib/main.js:1886:33)
    at ensureServiceIsRunning (/Users/marcotribuzio/WebstormProjects/simple-todos-vue/node_modules/esbuild/lib/main.js:2051:25)
    at build (/Users/marcotribuzio/WebstormProjects/simple-todos-vue/node_modules/esbuild/lib/main.js:1942:26)
    at bundleConfigFile (file:///Users/marcotribuzio/WebstormProjects/simple-todos-vue/node_modules/vite/dist/node/chunks/dep-3e87c7b2.js:63056:26)
    at loadConfigFromFile (file:///Users/marcotribuzio/WebstormProjects/simple-todos-vue/node_modules/vite/dist/node/chunks/dep-3e87c7b2.js:63032:31)
    at resolveConfig (file:///Users/marcotribuzio/WebstormProjects/simple-todos-vue/node_modules/vite/dist/node/chunks/dep-3e87c7b2.js:62656:34)
    at createViteServer (file:///Users/marcotribuzio/WebstormProjects/simple-todos-vue/node_modules/meteor-vite/dist/bin/worker.mjs:2463:22)
    at vite.server.start (file:///Users/marcotribuzio/WebstormProjects/simple-todos-vue/node_modules/meteor-vite/dist/bin/worker.mjs:2430:26)
    at async handleMessage (file:///Users/marcotribuzio/WebstormProjects/simple-todos-vue/node_modules/meteor-vite/dist/bin/worker.mjs:2552:3)

**Comment by MarcoTribuz** [comment]
Finally i resolved my issues with my mac, and i can confirm that the solution is working well :) @nachocodoner

**Comment by nachocodoner** [comment]
Thank you, @MarcoTribuz, for your input.

I was about to ask for more details about your setup and test your app without my changes to confirm whether they caused the issue. From the error logs, I noticed you were using Vite as well, that could have influenced the error. I'm glad to see the issue wasn’t related to my changes and that it’s working for you now. With the confirmations I’ve received so far, it seems like a good idea to include this fix in 3.1.

**Comment by nachocodoner** [comment]
Official [Meteor 3.1](https://dev.to/meteor/meteorjs-31-a-new-dawn-for-full-stack-javascript-development-4a13) has been released, including this Cordova issue mentioned here.

Update your app and share any feedback.

```shell
meteor update --release 3.1
```
